Core/HLE/SocketManager.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Core/HLE/SocketManager.h"
+#include "Common/Log.h"
+
+#include <mutex>
+
+// We use this array from 1 and forward. It's probably not a good idea to return 0 as a socket.
+InetSocket g_inetSockets[256];
+static std::mutex g_socketMutex;  // TODO: Remove once the adhoc thread is gone
+
+int AllocInetSocket() {
+	std::lock_guard<std::mutex> guard(g_socketMutex);
+	for (int i = MIN_VALID_INET_SOCKET; i < ARRAY_SIZE(g_inetSockets); i++) {
+		if (g_inetSockets[i].state == SocketState::Unused) {
+			return i;
+		}
+	}
+	_dbg_assert_(false);
+	ERROR_LOG(Log::sceNet, "Ran out of socket handles! This is BAD.");
+	return 0;
```

Comment:
Shouldn't we returned -1 in the case of failure to create a socket? just like a posix `socket` syscall:
> On success, a file descriptor for the new socket is returned.  On
       error, -1 is returned, and [errno](https://man7.org/linux/man-pages/man3/errno.3.html) is set to indicate the error.

Reply:
The caller is responsible for the actual syscall return value. But yeah this will be made clearer in an upcoming commit.

(either way, haven't hit this yet)